### PR TITLE
Add dynamic validator for field group

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix ?
* Ticket : #4113
* Funded by : DFAT
* Description : 
   - Add feature to create validator in field group. Used in Field mapping tool and keywords wizard.
   - Fix python error when select no keywords found layer
field_mapping_validation

### CODE REVIEW

**Comment by myarjunar** [comment]
LGTM

File: safe/common/parameters/group_select_parameter_widget.py [hunk 1]
```diff
@@ -73,6 +73,14 @@ def __init__(self, parameter, parent=None):
                     'max', 1))
                 double_spin_box.setSingleStep(value.get('constraint', {}).get(
                     'step', 0.01))
```
Comment (myarjunar) [inline safe/common/parameters/group_select_parameter_widget.py]: We usually pull left when we can.

File: safe/common/parameters/validators.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+"""List of custom validations for parameter container.."""
```
Comment (Gustry) [inline safe/common/parameters/validators.py]: Double full stop ?

File: safe/common/parameters/validators.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    :param validation_message: The message if there is validation error.
+    :type validation_message: str
+
+    :param kwargs: Keywords Argument.
+    :type kwargs: dict
+
+    :returns: Dictionary of valid and message.
+    :rtype: dict
+
+    Note: The code is not the best I wrote, since there are two alternatives.
```
Comment (Gustry) [inline safe/common/parameters/validators.py]: 😝

**Pull request merged**